Modules/Loadable/Segmentations/MRMLDM/vtkMRMLSegmentationsDisplayableManager2D.cxx

```diff
@@ -825,7 +875,36 @@ void vtkMRMLSegmentationsDisplayableManager2D::vtkInternal::UpdateDisplayNodePip
       // Set outline properties and turn it off if not shown
       if (segmentOutlineVisible)
         {
-        pipeline->LabelOutline->SetInputConnection(pipeline->Reslice->GetOutputPort());
+
+        // Set the outline threshold from the ThresholdValue field if it exists
+        // If ThresholdValue is not specified, then do not perform thresholding
+        vtkSmartPointer<vtkDoubleArray> thresholdValue = vtkDoubleArray::SafeDownCast(
+          imageData->GetFieldData()->GetAbstractArray(vtkMRMLSegmentationsDisplayableManager2D::GetThresholdValueFieldName()));
+        if (thresholdValue)
+          {
+
+          // Pad the extent of the image by one voxel in each direction
+          int extent[6] = {0,-1,0,-1,0,-1};
+          identityImageData->GetExtent(extent);
+          for (int i=0; i<2; ++i)
+            {
+            --extent[2*i];
+            ++extent[2*i+1];
+            }
+          pipeline->ImageConstantPad->SetInputData(identityImageData);
+          pipeline->ImageConstantPad->SetOutputWholeExtent(extent);
```

Comment:
In some structures there are non-empty values on the edges of the image, which causes problems with interpolation. Padding the image with empty pixels fixes this issue.